File: Testimonies/replace_engine_links.js
const fs = require('fs');

const files = [
    'index.html',
    'the-engine/engine_nodes.html',
    'view-schema/view-schema.html',
    'testimony/master-matrix.html',
    'risktable/investor-risk-matrix.html',
    'about/about.html'
];

const ICONS = {
    '01': '<svg class="w-4 h-4 text-blue-600 flex-shrink-0" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"><path stroke-linecap="round" stroke-linejoin="round" d="M3 16.5v2.25A2.25 2.25 0 0 0 5.25 21h13.5A2.25 2.25 0 0 0 21 18.75V16.5M16.5 12 12 16.5m0 0L7.5 12m4.5 4.5V3" /></svg>',
    '02': '<svg class="w-4 h-4 text-blue-600 flex-shrink-0" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"><path stroke-linecap="round" stroke-linejoin="round" d="M12 3c2.755 0 5.455.232 8.083.678.533.09.917.556.917 1.096v1.044a2.25 2.25 0 0 1-.659 1.591l-5.432 5.432a2.25 2.25 0 0 0-.659 1.591v2.927a2.25 2.25 0 0 1-1.244 2.013L9.75 21v-6.568a2.25 2.25 0 0 0-.659-1.591L3.659 7.409A2.25 2.25 0 0 1 3 5.818V4.774c0-.54.384-1.006.917-1.096A48.32 48.32 0 0 1 12 3Z" /></svg>',
    '03': '<svg class="w-4 h-4 text-blue-600 flex-shrink-0" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"><path stroke-linecap="round" stroke-linejoin="round" d="M13.19 8.688a4.5 4.5 0 0 1 1.242 7.244l-4.5 4.5a4.5 4.5 0 0 1-6.364-6.364l1.757-1.757m13.35-.622 1.757-1.757a4.5 4.5 0 0 0-6.364-6.364l-4.5 4.5a4.5 4.5 0 0 0 1.242 7.244" /></svg>',
    '04': '<svg class="w-4 h-4 text-teal-500 flex-shrink-0" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"><path stroke-linecap="round" stroke-linejoin="round" d="M3.75 12h16.5m-16.5 3.75h16.5M3.75 19.5h16.5M5.625 4.5h12.75a1.875 1.875 0 0 1 0 3.75H5.625a1.875 1.875 0 0 1 0-3.75Z" /></svg>',
    '05': '<svg class="w-4 h-4 text-teal-500 flex-shrink-0" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"><path stroke-linecap="round" stroke-linejoin="round" d="M2.25 18 9 11.25l4.306 4.306a11.95 11.95 0 0 1 5.814-5.518l2.74-1.22m0 0-5.94-2.281m5.94 2.28-2.28 5.941" /></svg>',
    '06': '<svg class="w-4 h-4 text-violet-600 flex-shrink-0" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"><path stroke-linecap="round" stroke-linejoin="round" d="M8.25 3v1.5M4.5 8.25H3m18 0h-1.5M4.5 12H3m18 0h-1.5m-15 3.75H3m18 0h-1.5M8.25 19.5V21M12 3v1.5m0 15V21m3.75-18v1.5m0 15V21m-9-1.5h10.5a2.25 2.25 0 0 0 2.25-2.25V6.75a2.25 2.25 0 0 0-2.25-2.25H6.75A2.25 2.25 0 0 0 4.5 6.75v10.5a2.25 2.25 0 0 0 2.25 2.25Z" /></svg>',
    '07': '<svg class="w-4 h-4 text-violet-600 flex-shrink-0" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"><path stroke-linecap="round" stroke-linejoin="round" d="M20.25 8.511c.884.284 1.5 1.128 1.5 2.097v4.286c0 1.136-.847 2.1-1.98 2.193-.34.027-.68.052-1.02.072v3.091l-3-3c-1.354 0-2.694-.055-4.02-.163a2.115 2.115 0 0 1-.825-.242m9.345-8.334a2.126 2.126 0 0 0-.476-.095 48.64 48.64 0 0 0-8.048 0c-1.131.094-1.976 1.057-1.976 2.192v4.286c0 .837.46 1.58 1.155 1.951m9.345-8.334V6.637c0-1.621-1.152-3.026-2.76-3.235A48.455 48.455 0 0 0 11.25 3c-2.115 0-4.198.137-6.24.402-1.608.209-2.76 1.614-2.76 3.235v6.226c0 1.621 1.152 3.026 2.76 3.235.577.075 1.157.14 1.74.194V21l4.155-4.155" /></svg>',
    '08': '<svg class="w-4 h-4 text-slate-500 flex-shrink-0" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"><path stroke-linecap="round" stroke-linejoin="round" d="M9 12.75 11.25 15 15 9.75m-3-7.036A11.959 11.959 0 0 1 3.598 6 11.99 11.99 0 0 0 3 9.749c0 5.592 3.824 10.29 9 11.623 5.176-1.332 9-6.03 9-11.622 0-1.31-.21-2.571-.598-3.751h-.152c-3.196 0-6.1-1.248-8.25-3.285Z" /></svg>',
    '09': '<svg class="w-4 h-4 text-amber-500 flex-shrink-0" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"><path stroke-linecap="round" stroke-linejoin="round" d="M15.75 5.25a3 3 0 0 1 3 3m3 0a6 6 0 0 1-7.029 5.912c-.563-.097-1.159.026-1.563.43L10.5 17.25H8.25v2.25H6v2.25H2.25v-2.818c0-.597.237-1.17.659-1.591l6.499-6.499c.404-.404.527-1 .43-1.563A6 6 0 1 1 21.75 8.25Z" /></svg>',
    '10': '<svg class="w-4 h-4 text-violet-600 flex-shrink-0" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"><path stroke-linecap="round" stroke-linejoin="round" d="m3.75 13.5 10.5-11.25L12 10.5h8.25L9.75 21.75 12 13.5H3.75Z" /></svg>',
    '11': '<svg class="w-4 h-4 text-red-500 flex-shrink-0" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"><path stroke-linecap="round" stroke-linejoin="round" d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-3L13.732 4c-.77-1.333-2.694-1.333-3.464 0L3.34 16c-.77 1.333.192 3 1.732 3Z" /></svg>'
};

// Replace regex
for (const file of files) {
    if (!fs.existsSync(file)) continue;
    let content = fs.readFileSync(file, 'utf8');

    // Desktop replacement map
    for (let c = 1; c <= 11; c++) {
        let i = c.toString().padStart(2, '0');
        let oldDesktop = `<div class="w-2 h-2 rounded-full ${c<=3 ? 'bg-blue-600' : c<=5 ? 'bg-teal-500' : c<=7 ? 'bg-violet-600' : c===8 ? 'bg-slate-500' : c===9 ? 'bg-amber-500' : c===10 ? 'bg-violet-600' : 'bg-red-500'}"></div>`;
        content = content.replace(new RegExp(oldDesktop, 'g'), ICONS[i]);
    }
    
    // Mobile replacement map (adds the icons before the texts)
    const texts = [
        "Data Ingest", "Null Triage", "Multi-Provider Enrichers", "L2 Order Book Snapshot",
        "Options Greeks Engine", "Regime Detection AI", "AI Sentiment Scrape", "Correlation Guard",
        "Confidence Gate", "RL Execution Node", "Circuit Breaker"
    ];
    
    for (let c = 1; c <= 11; c++) {
        let i = c.toString().padStart(2, '0');
        let txt = texts[c-1];
        let oldMobile = `→ ${txt}`;
        let newMobile = `${ICONS[i]} ${txt}`;
        content = content.replace(new RegExp(oldMobile, 'g'), newMobile);
    }
    
    // Mobile classes adjustment
    content = content.replace(/class="mobile-sub-link"/g, 'class="mobile-sub-link flex items-center gap-2"');
    content = content.replace(/class="py-2 border-b border-slate-50 text-slate-500 no-underline text-xs"/g, 'class="py-2 border-b border-slate-50 text-slate-500 no-underline text-xs flex items-center gap-2"');

    fs.writeFileSync(file, content);
}
console.log('Success.')

File: Testimonies/replace_engine_links.py
import os

html_files = [
    r"c:\Users\saviour\Documents\Wilbak\Testimonies\index.html",
    r"c:\Users\saviour\Documents\Wilbak\Testimonies\view-schema\view-schema.html",
    r"c:\Users\saviour\Documents\Wilbak\Testimonies\testimony\master-matrix.html",
    r"c:\Users\saviour\Documents\Wilbak\Testimonies\risktable\investor-risk-matrix.html",
    r"c:\Users\saviour\Documents\Wilbak\Testimonies\about\about.html"
]

def generate_desktop_dropdown(path_prefix, is_about=False):
    extra_class = " hover:text-slate-900 flex items-center gap-1 group no-underline transition-colors uppercase tracking-widest" if is_about else " flex items-center gap-1 group"
    return f'''                <div class="dropdown">
                    <a href="{path_prefix}the-engine/engine_nodes.html" class="dropdown-trigger{' ' if is_about else ' nav-link'}{extra_class}">
                        <span>The Engine</span>
                        <svg class="dropdown-arrow w-3 h-3 cursor-pointer group-hover:translate-y-0.5 transition-transform" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M19 9l-7 7-7-7"></path></svg>
                    </a>
                    <div class="dropdown-content">
                        <a href="{path_prefix}the-engine/engine_nodes.html#node-01" class="dropdown-item"><div class="w-2 h-2 rounded-full bg-blue-600"></div><span{' class="normal-case"' if is_about else ''}>Data Ingest</span></a>
                        <a href="{path_prefix}the-engine/engine_nodes.html#node-02" class="dropdown-item"><div class="w-2 h-2 rounded-full bg-blue-600"></div><span{' class="normal-case"' if is_about else ''}>Null Triage</span></a>
                        <a href="{path_prefix}the-engine/engine_nodes.html#node-03" class="dropdown-item"><div class="w-2 h-2 rounded-full bg-blue-600"></div><span{' class="normal-case"' if is_about else ''}>Multi-Provider Enrichers</span></a>
                        <a href="{path_prefix}the-engine/engine_nodes.html#node-04" class="dropdown-item"><div class="w-2 h-2 rounded-full bg-teal-500"></div><span{' class="normal-case"' if is_about else ''}>L2 Order Book Snapshot</span></a>
                        <a href="{path_prefix}the-engine/engine_nodes.html#node-05" class="dropdown-item"><div class="w-2 h-2 rounded-full bg-teal-500"></div><span{' class="normal-case"' if is_about else ''}>Options Greeks Engine</span></a>
                        <a href="{path_prefix}the-engine/engine_nodes.html#node-06" class="dropdown-item"><div class="w-2 h-2 rounded-full bg-violet-600"></div><span{' class="normal-case"' if is_about else ''}>Regime Detection AI</span></a>
                        <a href="{path_prefix}the-engine/engine_nodes.html#node-07" class="dropdown-item"><div class="w-2 h-2 rounded-full bg-violet-600"></div><span{' class="normal-case"' if is_about else ''}>AI Sentiment Scrape</span></a>
                        <a href="{path_prefix}the-engine/engine_nodes.html#node-08" class="dropdown-item"><div class="w-2 h-2 rounded-full bg-slate-500"></div><span{' class="normal-case"' if is_about else ''}>Correlation Guard</span></a>
                        <a href="{path_prefix}the-engine/engine_nodes.html#node-09" class="dropdown-item"><div class="w-2 h-2 rounded-full bg-amber-500"></div><span{' class="normal-case"' if is_about else ''}>Confidence Gate</span></a>
                        <a href="{path_prefix}the-engine/engine_nodes.html#node-10" class="dropdown-item"><div class="w-2 h-2 rounded-full bg-violet-600"></div><span{' class="normal-case"' if is_about else ''}>RL Execution Node</span></a>
                        <a href="{path_prefix}the-engine/engine_nodes.html#node-11" class="dropdown-item"><div class="w-2 h-2 rounded-full bg-red-500"></div><span{' class="normal-case"' if is_about else ''}>Circuit Breaker</span></a>
                    </div>
                </div>'''

def generate_mobile_dropdown(path_prefix, is_about=False):
    classes = "py-4 border-b border-slate-100 text-slate-900 no-underline" if is_about else "mobile-nav-link"
    link_class = "py-2 border-b border-slate-50 text-slate-500 no-underline text-xs" if is_about else "mobile-sub-link"
    wrapper = "pl-4 flex flex-col gap-1 pb-2 border-b border-slate-100" if is_about else "mobile-sub-nav"
    return f'''            <a href="{path_prefix}the-engine/engine_nodes.html" class="{classes}">The Engine</a>
            <div class="{wrapper}">
                <a href="{path_prefix}the-engine/engine_nodes.html#node-01" class="{link_class}">→ Data Ingest</a>
                <a href="{path_prefix}the-engine/engine_nodes.html#node-02" class="{link_class}">→ Null Triage</a>
                <a href="{path_prefix}the-engine/engine_nodes.html#node-03" class="{link_class}">→ Multi-Provider Enrichers</a>
                <a href="{path_prefix}the-engine/engine_nodes.html#node-04" class="{link_class}">→ L2 Order Book Snapshot</a>
                <a href="{path_prefix}the-engine/engine_nodes.html#node-05" class="{link_class}">→ Options Greeks Engine</a>
                <a href="{path_prefix}the-engine/engine_nodes.html#node-06" class="{link_class}">→ Regime Detection AI</a>
                <a href="{path_prefix}the-engine/engine_nodes.html#node-07" class="{link_class}">→ AI Sentiment Scrape</a>
                <a href="{path_prefix}the-engine/engine_nodes.html#node-08" class="{link_class}">→ Correlation Guard</a>
                <a href="{path_prefix}the-engine/engine_nodes.html#node-09" class="{link_class}">→ Confidence Gate</a>
                <a href="{path_prefix}the-engine/engine_nodes.html#node-10" class="{link_class}">→ RL Execution Node</a>
                <a href="{path_prefix}the-engine/engine_nodes.html#node-11" class="{link_class}">→ Circuit Breaker</a>
            </div>'''
            
for file in html_files:
    try:
        with open(file, 'r', encoding='utf-8') as f:
            content = f.read()
    except: continue
    
    path_prefix = "" if "index.html" in file else "../"
    is_about = "about.html" in file
    
    # Desktop
    if is_about:
        target_desktop = f'<a href="../the-engine/engine_nodes.html" class="hover:text-slate-900 no-underline transition-colors">The Engine</a>'
    else:
        target_desktop = f'<a href="{path_prefix}the-engine/engine_nodes.html" class="nav-link">The Engine</a>'
        
    if target_desktop in content:
        content = content.replace(target_desktop, generate_desktop_dropdown(path_prefix, is_about=is_about))
        
    # Mobile
    if is_about:
        target_mobile = f'<a href="../the-engine/engine_nodes.html" class="py-4 border-b border-slate-100 text-slate-900 no-underline">The Engine</a>'
    else:
        target_mobile = f'<a href="{path_prefix}the-engine/engine_nodes.html" class="mobile-nav-link">The Engine</a>'
        
    if target_mobile in content:
        content = content.replace(target_mobile, generate_mobile_dropdown(path_prefix, is_about=is_about))
        
    with open(file, 'w', encoding='utf-8') as f:
        f.write(content)

print("Done replacing.")
